### ISSUE TITLE
Feat: Support for Capacitor V6.0

### ISSUE THREAD

**Issue by xpluscal** [moberwasserlechner/capacitor-sms#24]
Is this currently planned?

**Comment by moberwasserlechner** [comment]
in 2 weeks hopefully I have the time todo it

**Comment by xpluscal** [comment]
@moberwasserlechner thank you for the response!
Is this going to be a huge lift?

**Comment by lajith111** [comment]
@xpluscal @moberwasserlechner  does this package automatically send SMS without default SMS APP? When i tried it opens in SMS APP. Thanks

**Comment by moberwasserlechner** [comment]
No. The default app always opens. Both platforms do not allow sending sms in the background since a few years.On 12.07.2024, at 06:22, lajith111 ***@***.***> wrote:﻿
@xpluscal @moberwasserlechner  does this package automatically send SMS without default SMS APP? When i tried it opens in SMS APP. Thanks

**Comment by lajith111** [comment]
@moberwasserlechner Thank you for quick reply,, Actually i am using this Cordova capacitor  plugin https://github.com/cordova-sms/cordova-sms-plugin

In this We have using below code for send SMS with target android 13  which works in Android 14 without opening default SMS app. This plugin if    //intent: '' it send sms automatically   but it require only SEND SMS permission

Apologize if i am `wrong` i am not android dev i am expertise in only .NET

```
var app = {
    sendSms: function() {
        var number = document.getElementById('numberTxt').value.toString(); /* iOS: ensure number is actually a string */
        var message = document.getElementById('messageTxt').value;
        console.log("number=" + number + ", message= " + message);

        //CONFIGURATION
        var options = {
            replaceLineBreaks: false, // true to replace \n by a new line, false by default
            android: {

                //intent: '' // send SMS without opening any other app, require : android.permission.SEND_SMS
            }
        };

        var success = function () { alert('Message sent successfully'); };
        var error = function (e) { alert('Message Failed:' + e); };
        sms.send(number, message, options, success, error);
    }
};
```

**Comment by moberwasserlechner** [comment]
Does your use case allows background sending and is your application live?

I dropped support because Android did not allow the background sending a few years back for most use cases. And less code means less maintenance work.

https://support.google.com/googleplay/android-developer/answer/10208820?sjid=3031348539978350994-EU

TL;DR: Imo this means: Only apps, which have SMS sending as their "core functionality/main purpose" are allowed to use background sending.

**Comment by lajith111** [comment]
Soyry for wasting yourtime.. Actually my App not in Play store / manually APK installation to users.
And my app does not use background for sending SMS and 

In our app users do the transaction & sent sms to customers using SIM card..
For this use case it will be helpful to us if you provide a option Thanks..

**Comment by moberwasserlechner** [comment]
Sry but such non standard use cases I can not support because of maintenance costs.

I have very little time for Open Soruce, which needs to be used efficiently. 

You can of course always fork this repo and use your own version of it.

**Comment by lajith111** [comment]
ok,Thanks

**Comment by moberwasserlechner** [comment]
Support for Capacitor 6 was release today. I could not test it on a device so please report any issues. I'll fix the Cap 6 related next week.
